extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -370,12 +370,11 @@ protected function createKeyForMailNotification( $watcher, WallNotificationEntit
 	}
 
 	protected function sendEmails( array $watchers, WallNotificationEntity $notification ) {
-		$text = strip_tags( $notification->data_noncached->msg_text, '<p><br>' );
+		$text = preg_replace( array( '#<br\s*/?>#i', '#</p\s*/?>#i' ), "\n",
+			$notification->data_noncached->msg_text );
+		$text = trim( strip_tags( $text ) );
 		$text = substr( $text, 0, 3000 ) . ( strlen( $text ) > 3000 ? '...' : '' );
-
-		$textNoHtml = preg_replace( '#<br\s*/?>#i', "\n", $text );
-		$textNoHtml = trim( preg_replace( '#</?p\s*/?>#i', "\n", $textNoHtml ) );
-		$textNoHtml = substr( $textNoHtml, 0, 3000) . ( strlen( $textNoHtml ) > 3000 ? '...' : '' );
+		$textNoNewlines = str_replace( "\n", ' ', $text );
```

Comment:
The '<p><br>' string is the list of allowable tags for the ```strip_tags``` function (https://php.net/strip_tags), so it was intentional for the <p> tag to be in there.  Was this causing display problems in the email?